src/bun.js/bindings/CallSitePrototype.cpp

```diff
@@ -0,0 +1,228 @@
+{
+    ENTER_PROTO_FUNC();
+    return JSC::JSValue::encode(callSite->functionName());
+}
+
+// TODO
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetMethodName, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    return callSiteProtoFuncGetFunctionName(globalObject, callFrame);
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetFileName, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+    return JSC::JSValue::encode(callSite->sourceURL());
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetLineNumber, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+    return JSC::JSValue::encode(callSite->lineNumber());
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetColumnNumber, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+    return JSC::JSValue::encode(callSite->columnNumber());
+}
+
+// TODO:
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetEvalOrigin, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    return JSC::JSValue::encode(JSC::jsUndefined());
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncGetScriptNameOrSourceURL, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+    return JSC::JSValue::encode(callSite->sourceURL());
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncIsToplevel, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+
+    JSC::JSValue thisValue = callSite->thisValue();
+
+    // This is what v8 does (JSStackFrame::IsToplevel in messages.cc):
+
+    if (thisValue.isUndefinedOrNull()) {
+        return JSC::JSValue::encode(JSC::jsBoolean(true));
+    }
+
+    JSC::JSObject* thisObject = thisValue.getObject();
+    if (thisObject && thisObject->isGlobalObject()) {
+        return JSC::JSValue::encode(JSC::jsBoolean(true));
+    }
+
+    return JSC::JSValue::encode(JSC::jsBoolean(false));
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncIsEval, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+
+    bool isEval = callSite->isEval();
+    return JSC::JSValue::encode(JSC::jsBoolean(isEval));
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncIsNative, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+
+    bool isNative = callSite->isNative();
+    return JSC::JSValue::encode(JSC::jsBoolean(isNative));
+}
+
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncIsConstructor, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+
+    bool isConstructor = callSite->isConstructor();
+    return JSC::JSValue::encode(JSC::jsBoolean(isConstructor));
+}
+
+// TODO:
+JSC_DEFINE_HOST_FUNCTION(callSiteProtoFuncIsAsync, (JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    ENTER_PROTO_FUNC();
+
+    return JSC::JSValue::encode(JSC::jsBoolean(false));
```

Comment:
This one I think can check if the `callee` `inherits` from `globalObject->asyncFunctionPrototype()`, maybe